src/main/java/org/broadinstitute/hellbender/tools/diagnostics/PrintBGZFBlockInformation.java

```diff
@@ -0,0 +1,189 @@
+        summary = "Print information about the compressed blocks in a BGZF format file",
+        oneLineSummary = "Print information about the compressed blocks in a BGZF format file",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class PrintBGZFBlockInformation extends CommandLineProgram {
+    private final Logger logger = LogManager.getLogger(PrintBGZFBlockInformation.class);
+
+    @Argument(fullName = "bgzf-file", doc = "The BGZF-format file for which to print block information", optional = false)
+    private String bgzfPathString;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "File to which to write block information (if not specified, prints to standard output", optional = true)
+    private String output;
+
+    private Path bgzfPath;
+
+    private long streamOffset = 0l;
+
+    private PrintStream outStream;
+
+    @Override
+    protected void onStartup() {
+        super.onStartup();
+
+        bgzfPath = IOUtils.getPath(bgzfPathString);
+
+        if ( ! Files.exists(bgzfPath) ) {
+            throw new UserException("File " + bgzfPathString + " does not exist");
+        }
+
+        if ( ! AbstractFeatureReader.hasBlockCompressedExtension(bgzfPathString) ) {
+            throw new UserException("File " + bgzfPathString + " does not end in a recognized BGZF file extension (" +
+                    AbstractFeatureReader.BLOCK_COMPRESSED_EXTENSIONS + ")");
+        }
+
+        if ( output != null ) {
+            try {
+                outStream = new PrintStream(output);
+            } catch (FileNotFoundException e) {
+                throw new UserException.CouldNotCreateOutputFile(output, "Unable to open output file", e);
+            }
+        } else {
+            outStream = System.out;
+        }
+    }
+
+    @Override
+    protected Object doWork() {
+        try ( InputStream bgzfInputStream = Files.newInputStream(bgzfPath) ) {
+            BGZFBlockMetadata blockInfo = null;
+            while ( (blockInfo = processNextBlock(bgzfInputStream, bgzfPathString)) != null ) {
+                outStream.println(String.format("Block at file offset %d", blockInfo.blockOffset));
+                outStream.println(String.format("\t- compressed size: %d", blockInfo.compressedSize));
+                outStream.println(String.format("\t- uncompressed size: %d", blockInfo.uncompressedSize));
+                outStream.println();
+            }
+        } catch ( IOException e ) {
+            throw new UserException("Error while parsing BGZF file. Error message was: " + e.getMessage(), e);
+        }
+
+        return 0;
+    }
+
+    @Override
+    protected void onShutdown() {
+        if ( outStream != null && outStream != System.out ) {
+            outStream.close();
+        }
+    }
+
+    private BGZFBlockMetadata processNextBlock( InputStream stream, String streamSource) throws IOException {
```

Comment:
This looks like it was pulled from htsjdk, this should maybe exist in some common utility method somewhere exposed in htsjdk ideally.

Reply:
Ideally, yes, but beyond the scope of this PR.